boot/freeldr/freeldr/arch/i386/pcmem.c

```diff
@@ -286,6 +290,22 @@ PcMemGetBiosMemoryMap(PFREELDR_MEMORY_DESCRIPTOR MemoryMap, ULONG MaxMemoryMapSi
             break;
         }
 
+        /* Extra (input) safety. */
+        if (Regs.x.ecx < 20 || BiosMemoryMapSize < Regs.x.ecx)
```

Comment:
The magic 20 would make more sense as `RTL_SIZEOF_THROUGH_FIELD(BIOS_MEMORY_MAP, Type)` or `FIELD_OFFSET(BIOS_MEMORY_MAP, Reserved)`. That would also save you from having to C_ASSERT above.

Reply:
I replaced the magic number with a comment + define.
Actually, I do want to C_ASSERT without reference to BIOS_MEMORY_MAP, as I'm checking the latter against the interrupt requirement, as is OS vs BIOS/API.